.github/workflows/publish_docs.yml

```diff
@@ -0,0 +1,19 @@
+name: 'publish docs'
+
+# Publish when upon push to main or add-docs branches
+on:
+  push:
+    branches:
+      - main
+      - add-docs
+    tags:
+      - '*'
+
+jobs:
+  publish_sphinx_docs:
+    name: Publish Sphinx Docs
+    runs-on: ubuntu-latest
+    steps:
+    - uses: neuroinformatics-unit/actions/publish_sphinx_docs@v1
+      with:
+        secret_input: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
That's a neat solution. I will try to implement the two-job workflow upstream, in our actions-repo, since this issue affects all our other repos-with-docs as well.

Reply:
I implemented the new two job workflow upstream [in this PR](https://github.com/neuroinformatics-unit/actions/pull/13).
To test things, I made both the build and the deploy jobs run from this branch, and it seems to have worked.

Before merging the current PR, I will configure it so that all merges to main and all pull requests trigger the build action, but only pushing/merging to main triggers a deployment to gh-pages. In the future, we may change this to only be triggered by pushing release tags. Does that sound reasonable @samcunliffe ?

Reply:
All fine for me. I leave it for you to merge when ready.